### PR TITLE
Add support for sending images

### LINKED ISSUE #2: Add support for sending images
# Request

- As an author (a user sending a message) I want to be able to send images using the chat. I also want to be able to see the image in the chat, to verify that it worked.
- As a consumer (a user receiving a message) I want to be able to view an image sent to me in the chat.

# Constraints

For this first version the chat only has to work with small photos (<10kb).  See an example below.

# Example
image

### PR DESCRIPTION
**Pull request by manimul**
@hmps this pr offers a solution to the issue you submitted

### CODE REVIEW

File: packages/client/components/useChat.ts [hunk 1]
```diff
@@ -1,12 +1,12 @@
-import { useEffect, useRef, useState } from 'react';
-import socketIOClient from 'socket.io-client';
+import { useEffect, useRef, useState } from "react";
```
Comment (hmps) [inline packages/client/components/useChat.ts]: Could you please revert all the unrelated changes? Something with you editor seems to autoformat `'` to `"` for example.

File: packages/client/components/useChat.ts [hunk 2]
```diff
@@ -30,9 +30,10 @@ const useChat = (roomId: string): Chat => {
     };
   }, [roomId]);
 
-  const sendMessage = (messageBody: string) => {
+  const sendMessage = (messageBody: any) => {
```
Comment (hmps) [inline packages/client/components/useChat.ts]: Could we avoid falling back on `any` here?

File: packages/client/interfaces/index.ts [hunk 3]
```diff
@@ -6,10 +6,11 @@ export type User = {
...
   ownedByCurrentUser: boolean;
   body: string;
+  image: string;
   senderId: string;
 };
 
 export interface Chat {
   messages: Message[];
-  sendMessage: (m: string) => void;
+  sendMessage: (m: object) => void;
```
Comment (hmps) [inline packages/client/interfaces/index.ts]: What is the actual type of this `object`?

File: packages/client/pages/chat/[id].tsx [hunk 4]
```diff
@@ -1,38 +1,64 @@
...
+import styles from "./chat.module.css";
+import useChat from "../../components/useChat";
 
 const ChatRoom = (): ReactElement => {
   const router = useRouter();
   const { id: roomId } = router.query;
   const { messages, sendMessage } = useChat(roomId as string);
-  const [newMessage, setNewMessage] = useState('');
+  const [newMessage, setNewMessage] = useState("");
+  const [newMessageObj] = useState({} as any);
```
Comment (hmps) [inline packages/client/pages/chat/[id].tsx]: Why use `useState` if it's not being updated?

Comment (hmps) [inline packages/client/pages/chat/[id].tsx]: Could we avoid `any` here?

File: packages/client/pages/chat/[id].tsx [hunk 5]
```diff
@@ -1,38 +1,64 @@
...
+  const [newImage, setNewImage] = useState("");
+  const imageInputRef = React.useRef<HTMLInputElement>(null);
 
   const handleNewMessageChange = (event: ChangeEvent<HTMLTextAreaElement>) => {
     setNewMessage(event.target.value);
   };
 
+  const handleNewImageChange = (event: ChangeEvent<HTMLInputElement>) => {
+    setNewImage(URL.createObjectURL(event.target.files![0]));
```
Comment (hmps) [inline packages/client/pages/chat/[id].tsx]: Can you explain the usage of `!` here?

File: packages/client/components/useChat.ts (hunk 1 above)
Comment (manimul) [reply packages/client/components/useChat.ts]: Fixed this in latest commit to marks_branch

File: packages/client/components/useChat.ts (hunk 2 above)
Comment (manimul) [reply packages/client/components/useChat.ts]: I have updated my submission and used the Message object shape here, as it has the required properties. Is this a better approach?

File: packages/client/interfaces/index.ts (hunk 3 above)
Comment (manimul) [reply packages/client/interfaces/index.ts]: In the resubmission I am using the Message type here.

File: packages/client/pages/chat/[id].tsx (hunk 4 above)
Comment (manimul) [reply packages/client/pages/chat/[id].tsx]: I have changed this now. I don't know exactly why I did it tbh...I now use the Message object here, without State
`  const newMessageObj = {} as Message;
`

Comment (manimul) [reply packages/client/pages/chat/[id].tsx]: I have changed this to
`  const newMessageObj = {} as Message;
`
Is this a better aproach?

File: packages/client/pages/chat/[id].tsx (hunk 5 above)
Comment (manimul) [reply packages/client/pages/chat/[id].tsx]: This had me stumped. Struggling to understand why i did this too...
I know i was being prompted by eslint to safeguard against the Object being null, and oddly enough this removed the warning...but its not a conventional conditional syntax, right?
I have switched to using a standard if statement, is this okay practice?
`if (event.target.files != null) {
      setNewImage(URL.createObjectURL(event.target.files[0]));
    }`

**Review by hmps — APPROVED** [review]
LGTM! 👍
